### ISSUE TITLE
Increase MAX_QUIC_CONNECTIONS_PER_PEER for staked validators

### ISSUE THREAD

**Issue by SOELTH** [solana-labs/solana#29921]
#### Problem
In order to provide their clients with best-possible execution, experienced Validator/RPC providers often run their nodes behind NATS in order to provide staked sending to a larger clientele without overwhelming a single node. Additionally, single nodes can open multiple QUIC connections! Because of this limitation, opening staked connections and scaling staked sending becomes difficult. 

#### Proposed Solution
Increase MAX_QUIC_CONNECTIONS_PER_PEER for staked validators:

I am open to a flat-increase or a stake-weighted increase.

**Comment by bw-solana** [comment]
@SOELTH - While the number of connections is the same for staked/unstaked, the receive window of the connection is scaled based on stake weight. Is it okay to use this knob, or is there some reason we need to scale the actual number of connections?

@lijunwangs - What do you think?

**Comment by lijunwangs** [comment]
The receive window and max streams count are other knobs -- but they are enforced per connection. I think it is a legitimate ask to have more max connections per peer for staked vs non staked. @SOELTH can you elaborate more on the use case? Do you mean these  Validator/RPC itself is staked and they are serving large number of clients and forwarding transactions to the network using the same (IP, pubkey) pair? And you find the current 8 connections are not enough?

**Comment by SOELTH** [comment]
@lijunwangs Right, so the use case here is to to put the Validator and several other RPC nodes behind a NAT. This allows me and other Validator operators to offer our clients staked transaction forwarding without overloading our Validators, where performance is critical. The default RPC client opens several connections to each node, so currently I can only run a single node behind this NAT before I see connections being pruned because of this limit. 

TLDR, we want to offload transaction forwarding services to dedicated nodes we're hosting for our clients, but still get the benefit of staked transaction forwarding. With the current MAX_CONNECTIONS_PER_PEER limit this just isn't possible.

Obviously, there should be *SOME* kind of limit here, but imo the current is much too low. 

I'll note this is unrelated to window-size, which deals more with throughput than individual connections, which can't be shared by different nodes.

**Comment by lijunwangs** [comment]
@SOELTH Thanks for this explanation. Do these RPC nodes share the staked nodes' pubkey?

**Comment by t-nelson** [comment]
seems like the path forward here is to replace the constant with a cli arg.  let operators shoot themselves in the foot, don't force it on everyone

**Comment by mschneider** [comment]
@SOELTH we had similar issues before, you might want to check out this proxy we built, it's a standalone mirco-service that can handle the tx forwarding on behalf of the staked validator (by sharing identities) and at the same time multiplex packets of different clients into one connection. this way you won't run into max_connections_per_peer.

https://github.com/blockworks-foundation/lite-rpc
